blueprints/structural_sections/concrete/reinforced_concrete_sections/rectangular.py

```diff
@@ -0,0 +1,341 @@
+                shear_check=shear_check,
+                torsion_check=torsion_check,
+                mandrel_diameter_factor=mandrel_diameter_factor,
+                anchorage_length=anchorage_length,
+                based_on_cover=True,
+                relative_start_position=relative_start_position,
+                relative_end_position=relative_end_position,
+            )
+        )
+
+    def add_stirrup_in_center(
+        self,
+        width: MM,
+        diameter: MM,
+        distance: MM,
+        material: ReinforcementSteelMaterial,
+        shear_check: bool = True,
+        torsion_check: bool = True,
+        mandrel_diameter_factor: float | None = None,
+        anchorage_length: MM = 0.0,
+        relative_start_position: RATIO = 0.0,
+        relative_end_position: RATIO = 1.0,
+    ) -> StirrupConfiguration:
+        """Add stirrups to the center of the reinforced cross-section based on a given width (ctc of the legs). The created configuration goes around
+        the longitudinal rebars (if any).
+
+        Use .add_stirrup_configuration() to add a stirrup configuration of any shape, size, and position (as long as it is inside the cross-section).
+
+        Parameters
+        ----------
+        width: MM
+            Total width of the stirrup taken from the center lines of the legs [mm].
+        diameter: MM
+            Diameter of the stirrups [mm].
+        distance: MM
+            Longitudinal distance between stirrups [mm].
+        material : ReinforcementSteelMaterial
+            Representation of the properties of reinforcement steel suitable for use with NEN-EN 1992-1-1
+        shear_check: bool
+            Take stirrup into account in shear check
+        torsion_check: bool
+            Take stirrup into account in torsion check
+        mandrel_diameter_factor: DIMENSIONLESS
+            Inner diameter of mandrel as multiple of stirrup diameter [-]
+            (default: 4⌀ for ⌀<=16mm and 5⌀ for ⌀>16mm) Tabel 8.1Na NEN-EN 1992-1-1 Dutch National Annex.
+        anchorage_length: MM
+            Anchorage length [mm]
+        relative_start_position: RATIO
+            Relative position of the start of the stirrup configuration inside the cross-section (longitudinal direction). Value between 0 and 1.
+            Default is 0 (start).
+        relative_end_position: RATIO
+            Relative position of the end of the stirrup configuration inside the cross-section (longitudinal direction). Value between 0 and 1.
+            Default is 1 (end).
+
+        Returns
+        -------
+        StirrupConfiguration
+            Newly created stirrup configuration inside the cross-section.
+        """
+        # get the corners of the cross-section
+        _, min_y, _, max_y = self.cross_section.geometry.bounds
+
+        # create the corners of the stirrup configuration based on the covers present
+        left_bottom_corner = Point(-width / 2, min_y + self.covers.lower + (diameter / 2))
```

Comment:
https://github.com/Blueprints-org/blueprints/pull/287/files#r1644197316

Reply:
"this means that the stirrup ALWAYs is the outer layer of rebar, most of the times that is the case, but not always."

Yes, that is indeed the case. In the docstring I have added this to clarify:
"        
....
Use .add_stirrup_configuration() to add a stirrup configuration of any shape, size, and position (as long as it is inside the cross-section).

....
"